### PR TITLE
Refactor testdata_generator

### PR DESCRIPTION
**Pull request by rbuffat**
Refactor test function of testdata_generator:
- Test a collection of records instead of only single records and do not assume any order (resolves https://github.com/Toblerity/Fiona/issues/933)
- Only compare ints (should resolve https://github.com/Toblerity/Fiona/issues/935 )

### CODE REVIEW

**Comment by rbuffat** [comment]
Would it make sense that all tests for drvsupport.py using the testgenerator run only on the CI or with a special marker?

**Comment by QuLogic** [comment]
I attempted to rebase this to apply to 1.8.18, but unfortunately, it does not appear to fix #935. The logging says "Could not extract unique elements from test_data."

File: tests/conftest.py [hunk 1]
```diff
@@ -395,18 +396,38 @@ def get_records2(driver, range):
...
+            logging.error("Could not extract unique elements from test_data.")
+            return False
+
+        for test_record in test_data:
+            pos = _get_pos(driver, test_record)
+            if pos not in positions:
+                logging.error("Test record {} corresponds not to a record from true_data.".format(test_record))
+                return False
+
+        return True
```
Comment (QuLogic) [inline tests/conftest.py:430]: You should just `assert` these things and let pytest print out the parts that don't match.

Comment (rbuffat) [reply tests/conftest.py:430]: The assert is in the corresponding tests.

Comment (QuLogic) [reply tests/conftest.py:430]: It's less effective; the tests will just `assert False` and that's all pytest will show. Then you have to go through the log to figure out what from here failed. If you assert here, then pytest will show the individual parts of what are not matching, and the backtrace will point to the actual failure.

Comment (snorfalorpagus) [reply tests/conftest.py:430]: I agree with @QuLogic. We should let pytest do the work for us.

```python
assert len(actual_data) == len(expected_data)
```

Gives:

```
    def test_me():
        expected_data = [1,2,3]
        actual_data = [1,2]
>       assert len(actual_data) == len(expected_data)
E       assert 2 == 3
E        +  where 2 = len([1, 2])
E        +  and   3 = len([1, 2, 3])

tests/test_me.py:4: AssertionError
```

**Comment by sgillies** [comment]
@rbuffat I'm going to reject this one for 1.8 and 1.9.

**Pull request closed without merging**